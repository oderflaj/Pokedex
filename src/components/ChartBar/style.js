import {StyleSheet} from 'react-native';

export const ChartStyle = new StyleSheet.create({
  containerChart: {
    width: '80%',
    alignItems: 'center',
    justifyContent: 'flex-start',
    flexDirection: 'row',
  },
  blueLine: {
    borderTopLeftRadius: 3,
    borderBottomLeftRadius: 3,
    height: 7,
    backgroundColor: '#0D69B8',
  },

  greyLine: {
    borderTopRightRadius: 3,
    borderBottomRightRadius: 3,
    height: 7,
    backgroundColor: '#DAD7D7',
  },
  containerText: {
    width: 35,
    justifyContent: 'center',
    alignItems: 'center',
    borderWidth: 0.5,
    borderRadius: 3,
    borderColor: '#D5D6D7',
  },
  text: {
    fontSize: 12,
    fontWeight: 'bold',
  },
});
