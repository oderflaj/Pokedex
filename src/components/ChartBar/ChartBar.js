import React from 'react';
import {View, Text} from 'react-native';
import {ChartStyle} from './style';

export default ({number}) => {
  const blueSize = `${
    number > 90 ? (number > 100 ? 90 : number - 15) : number - 10
  }%`;
  const greySize = `${100 - number - 10}%`;
  return (
    <View style={[ChartStyle.containerChart]}>
      <View style={[ChartStyle.blueLine, {width: blueSize}]}></View>
      <View style={[ChartStyle.containerText]}>
        <Text style={ChartStyle.text}>{number}</Text>
      </View>
      <View style={[ChartStyle.greyLine, {width: greySize}]}></View>
    </View>
  );
};
