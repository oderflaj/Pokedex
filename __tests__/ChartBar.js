/* global __DEV__ */
import 'react-native';
import React from 'react';
import ChartBar from '../src/components/ChartBar/ChartBar';
import renderer from 'react-test-renderer';

describe('<ChartBar/>', () => {
  it('Test to render correctly with a value between 0 and 85', () => {
    const ChartBarData = renderer.create(<ChartBar number={150} />).toJSON();
    expect(ChartBarData).toMatchSnapshot();
  });

  it('It is getting an equal result of maximum percentage using two different values in 2 components: 101 and 126', () => {
    const ChartBarData1 = renderer.create(<ChartBar number={101} />);
    //console.log(ChartBarData1.toJSON().children[0].props.style[1]);

    const ChartBarData2 = renderer.create(<ChartBar number={126} />);
    //console.log(ChartBarData2);

    expect(ChartBarData1.toJSON().children[0].props.style[1].width).toBe(
      ChartBarData2.toJSON().children[0].props.style[1].width,
    );
  });

  it('It is getting a different result of maximum and normal percentage using two different values in 2 components: 55 and 126', () => {
    const ChartBarData1 = renderer.create(<ChartBar number={55} />);
    expect(ChartBarData1.toJSON().children[0].props.style[1].width).toBe('45%');

    const ChartBarData2 = renderer.create(<ChartBar number={126} />);
    expect(ChartBarData2.toJSON().children[0].props.style[1].width).toBe('90%');
  });

  it('It is geting value less 15 for percentage between 90% and 100%, for 97 the value is 82%', () => {
    const ChartBarData1 = renderer.create(<ChartBar number={97} />);
    expect(ChartBarData1.toJSON().children[0].props.style[1].width).toBe('82%');
  });

  it('It is geting always 90% when the percentage is over 100%, for 107 it is getting 90%', () => {
    const ChartBarData1 = renderer.create(<ChartBar number={107} />);
    expect(ChartBarData1.toJSON().children[0].props.style[1].width).toBe('90%');
  });
});
