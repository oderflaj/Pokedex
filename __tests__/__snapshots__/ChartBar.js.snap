// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartBar/> Test to render correctly with a value between 0 and 85 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "width": "80%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#0D69B8",
          "borderBottomLeftRadius": 3,
          "borderTopLeftRadius": 3,
          "height": 7,
        },
        Object {
          "width": "90%",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#D5D6D7",
          "borderRadius": 3,
          "borderWidth": 0.5,
          "justifyContent": "center",
          "width": 35,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "fontSize": 12,
          "fontWeight": "bold",
        }
      }
    >
      150
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#DAD7D7",
          "borderBottomRightRadius": 3,
          "borderTopRightRadius": 3,
          "height": 7,
        },
        Object {
          "width": "-60%",
        },
      ]
    }
  />
</View>
`;
